src/components/PoolingWidget/DefineSpread.tsx

```diff
@@ -35,17 +39,69 @@ export const SpreadInformation: React.FC<SpreadInformationProps> = ({ selectedTo
   )
 }
 
-const DefineSpread: React.FC<DefineSpreadProps> = ({ selectedTokensMap, spread, setSpread }) => {
-  const handleSpreadChange = ({ target: { value } }: React.ChangeEvent<HTMLInputElement>): void => {
-    if (+value < 0 || +value >= 100) return
+const DefineSpread: React.FC<DefineSpreadProps> = ({ selectedTokensMap, spread, setSpread, nextStep }) => {
+  const onChange = useCallback(
+    ({ currentTarget: { value } }: React.SyntheticEvent<HTMLInputElement>): void => {
+      const newValue = value
+        // put a `0` at the start, if starting with `.`
+        .replace(/^(\..*)$/, '0$1')
+        // remove extra decimal places, if present and more than 1
+        .replace(/^(.*\.\d{0,1}).*$/, '$1')
+
+      setSpread(Number(newValue))
+    },
+    [setSpread],
+  )
+
+  const onBlur = useCallback(
+    ({ currentTarget: { value } }: React.SyntheticEvent<HTMLInputElement>): void => {
+      let newValue
 
-    setSpread(+value)
-  }
+      // Adjust amount in case invalid
+      if (isNaN(+value) || +value <= 0) {
+        // set min value
+        newValue = '0.1'
+      } else if (+value >= 100) {
+        // set max value
+        newValue = '99.9'
+      } else {
+        newValue = adjustPrecision(value, 1)
+      }
+
+      setSpread(Number(newValue))
+    },
+    [setSpread],
+  )
+
+  const onKeyPress = useCallback(
+    (event: React.KeyboardEvent<HTMLInputElement>): void => {
+      if (event.key === 'Enter') {
+        onBlur(event)
+        nextStep()
+      } else if (
+        // allow to remove chars and add numbers and a dot
+        !event.key.match(/(Backspace|Delete|\.|[0-9])/) ||
+        // do not allow a second dot
+        (event.key === '.' && event.currentTarget.value.match(/\./))
```

Comment:
It is generally recommended to use `regexp.test` if you are only interested in `true|false`. And `string.includes` to check for 1 char only
```suggestion
        !/(Backspace|Delete|\.|[0-9])/.test(event.key) ||
        // do not allow a second dot
        (event.key === '.' && event.currentTarget.value.includes('.')
```